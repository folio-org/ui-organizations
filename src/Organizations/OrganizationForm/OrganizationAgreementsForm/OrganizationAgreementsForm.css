.panels {
  background: var(--panel-bg-color);
  padding: 1.2rem 1.2rem 0;
  border-radius: var(--panel-radius);
  border-bottom: var(--panel-border-bottom);
}
