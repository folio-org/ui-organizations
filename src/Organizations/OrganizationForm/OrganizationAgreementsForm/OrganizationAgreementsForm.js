import React, { Component } from 'react';
import { FormattedMessage } from 'react-intl';
import { Field, FieldArray } from 'redux-form';

import {
  Button,
  Col,
  IconButton,
  Row,
  TextArea,
  TextField,
} from '@folio/stripes/components';

import { Required, isURLValid } from '../../../Utils/Validate';
import css from './OrganizationAgreementsForm.css';

class OrganizationAgreementsForm extends Component {
  constructor(props) {
    super(props);
    this.renderForm = this.renderForm.bind(this);
    this.renderSubForm = this.renderSubForm.bind(this);
  }

  renderForm = ({ fields }) => {
    return (
      <Row>
        <Col xs={12}>
          {fields.length === 0 &&
            <Col xs={6}>
              <div><em>{<FormattedMessage id="ui-organizations.agreement.pleaseAddAgreements" />}</em></div>
            </Col>
          }
          {fields.map(this.renderSubForm)}
        </Col>
        <Col xs={12} style={{ paddingTop: '10px' }}>
          <Button
            data-test-vendor-term-add
            onClick={() => fields.push({})}
          >
            <FormattedMessage id="ui-organizations.agreement.add" />
          </Button>
        </Col>
      </Row>
    );
  }

  renderSubForm = (elem, index, fields) => {
    return (
      <div key={index} className={css.panels}>
        <Row key={index}>
          <Col
            xs={6}
            md={3}
          >
            <Field
              component={TextField}
              fullWidth
              id={`${elem}.name`}
              label={<FormattedMessage id="ui-organizations.agreement.name" />}
              name={`${elem}.name`}
              required
              validate={[Required]}
            />
          </Col>
          <Col
            xs={6}
            md={3}
          >
            <Field
              component={TextField}
              fullWidth
              id={`${elem}.discount`}
              label={<FormattedMessage id="ui-organizations.agreement.discountSym" />}
              name={`${elem}.discount`}
              type="number"
            />
          </Col>
          <Col
            xs={6}
            md={3}
          >
            <Field
              component={TextField}
              fullWidth
              id={`${elem}.referenceUrl`}
              label={<FormattedMessage id="ui-organizations.agreement.referenceUrl" />}
              name={`${elem}.referenceUrl`}
              type="text"
              validate={[isURLValid]}
            />
          </Col>
          <Col
            xs={5}
            md={2}
          >
            <Field
              component={TextArea}
              fullWidth
              id={`${elem}.notes`}
              label={<FormattedMessage id="ui-organizations.agreement.notes" />}
              name={`${elem}.notes`}
            />
          </Col>
          <Col
            xs={1}
            style={{ paddingTop: '20px' }}
          >
            <IconButton
              data-test-vendor-term-remove
              icon="trash"
              onClick={() => fields.remove(index)}
            >
              {<FormattedMessage id="ui-organizations.agreement.remove" />}
            </IconButton>
          </Col>
        </Row>
      </div>
    );
  }

  render() {
    return (
      <FieldArray
        component={this.renderForm}
        id="agreements"
        name="agreements"
      />

    );
  }
}

export default OrganizationAgreementsForm;
