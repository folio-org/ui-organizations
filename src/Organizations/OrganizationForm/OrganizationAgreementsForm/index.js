export { default as OrganizationAgreementsForm } from './OrganizationAgreementsForm';
