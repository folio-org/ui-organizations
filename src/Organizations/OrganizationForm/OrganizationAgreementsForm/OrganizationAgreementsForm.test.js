import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import stripesFinalForm from '@folio/stripes/final-form';

import OrganizationAgreementsForm from './OrganizationAgreementsForm';

const TestForm = stripesFinalForm({})(
  () => {
    return (
      <form>
        <OrganizationAgreementsForm />
      </form>
    );
  },
);

const agreement = {
  discount: 0,
  name: 'Full price',
  notes: '',
  referenceUrl: '',
};
const renderForm = ({ initialValues = {} } = {}) => render(
  <TestForm
    onSubmit={jest.fn()}
    initialValues={initialValues}
  />,
  { wrapper: MemoryRouter },
);

describe('OrganizationAgreementsForm', () => {
  beforeEach(() => {
    global.document.createRange = global.document.originalCreateRange;
  });

  afterEach(() => {
    global.document.createRange = global.document.mockCreateRange;
  });

  it('should render correct structure with defined accounts', async () => {
    const { asFragment } = renderForm({ initialValues: { agreements: [agreement] } });

    await screen.findByText('ui-organizations.agreement.name');

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correct structure without accounts', async () => {
    const { asFragment } = renderForm();

    expect(asFragment()).toMatchSnapshot();
  });
});
