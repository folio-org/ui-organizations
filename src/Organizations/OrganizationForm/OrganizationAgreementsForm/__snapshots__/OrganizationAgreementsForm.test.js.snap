// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganizationAgreementsForm should render correct structure with defined accounts 1`] = `
<DocumentFragment>
  <form>
    <div
      class="row"
    >
      <div
        class="col-xs-12"
      >
        <div
          class="panels"
        >
          <div
            class="row"
          >
            <div
              class="col-xs-6 col-md-3"
            >
              <div
                class="textField"
              >
                <label
                  class="label"
                  for="agreements[0].name"
                >
                  ui-organizations.agreement.name
                  <span
                    aria-hidden="true"
                    class="asterisk"
                  >
                    *
                  </span>
                </label>
                <div
                  class="formControl inputGroup"
                >
                  <input
                    aria-invalid="false"
                    aria-required="true"
                    autocapitalize="off"
                    autocomplete="off"
                    autocorrect="off"
                    error=""
                    id="agreements[0].name"
                    name="agreements[0].name"
                    required=""
                    spellcheck="false"
                    type="text"
                    value="Full price"
                    warning=""
                  />
                </div>
                <div
                  role="alert"
                />
              </div>
            </div>
            <div
              class="col-xs-6 col-md-3"
            >
              <div
                class="textField"
              >
                <label
                  class="label"
                  for="agreements[0].discount"
                >
                  ui-organizations.agreement.discountSym
                </label>
                <div
                  class="formControl inputGroup"
                >
                  <input
                    aria-invalid="false"
                    autocapitalize="off"
                    autocomplete="off"
                    autocorrect="off"
                    error=""
                    id="agreements[0].discount"
                    name="agreements[0].discount"
                    spellcheck="false"
                    type="number"
                    value="0"
                    warning=""
                  />
                </div>
                <div
                  role="alert"
                />
              </div>
            </div>
            <div
              class="col-xs-6 col-md-3"
            >
              <div
                class="textField"
              >
                <label
                  class="label"
                  for="agreements[0].referenceUrl"
                >
                  ui-organizations.agreement.referenceUrl
                </label>
                <div
                  class="formControl inputGroup"
                >
                  <input
                    aria-invalid="false"
                    autocapitalize="off"
                    autocomplete="off"
                    autocorrect="off"
                    error=""
                    id="agreements[0].referenceUrl"
                    name="agreements[0].referenceUrl"
                    spellcheck="false"
                    type="text"
                    value=""
                    warning=""
                  />
                </div>
                <div
                  role="alert"
                />
              </div>
            </div>
            <div
              class="col-xs-5 col-md-2"
            >
              <div
                class="textArea fullWidth"
              >
                <label
                  class="label"
                  for="agreements[0].notes"
                >
                  ui-organizations.agreement.notes
                </label>
                <div
                  class="inputGroup"
                >
                  <textarea
                    aria-invalid="false"
                    class="formControl"
                    id="agreements[0].notes"
                    name="agreements[0].notes"
                  />
                  <div
                    role="alert"
                  />
                </div>
              </div>
            </div>
            <div
              class="col-xs-1"
              style="padding-top: 20px;"
            >
              <button
                aria-label="trash"
                class="iconButton medium"
                data-test-vendor-term-remove="true"
                icon="trash"
                type="button"
              >
                <span
                  class="iconButtonInner mediumOffset"
                  data-test-vendor-term-remove="true"
                >
                  <span>
                    Icon
                  </span>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col-xs-12"
        style="padding-top: 10px;"
      >
        <button
          class="button default"
          data-test-vendor-term-add="true"
          type="button"
        >
          <span
            class="inner"
          >
            ui-organizations.agreement.add
          </span>
        </button>
      </div>
    </div>
  </form>
  <div />
</DocumentFragment>
`;

exports[`OrganizationAgreementsForm should render correct structure without accounts 1`] = `
<DocumentFragment>
  <form>
    <div
      class="row"
    >
      <div
        class="col-xs-12"
      >
        <div
          class="col-xs-6"
        >
          <div>
            <em>
              ui-organizations.agreement.pleaseAddAgreements
            </em>
          </div>
        </div>
      </div>
      <div
        class="col-xs-12"
        style="padding-top: 10px;"
      >
        <button
          class="button default"
          data-test-vendor-term-add="true"
          type="button"
        >
          <span
            class="inner"
          >
            ui-organizations.agreement.add
          </span>
        </button>
      </div>
    </div>
  </form>
  <div />
</DocumentFragment>
`;
